### DISCUSSION TITLE
config.json from URL

### DISCUSSION THREAD

**Discussion by johannesnicolaus** [GMOD/jbrowse-components#4000] [Q&A] [answered]
Hi, I'm running jbrowse2 using npm. I'm wondering if it is possible to load the `config.json` file from URL as I want to have the config file on github and not have to replace the local file every time I do updates.

**Comment by cmdcolin** [comment]
@johannesnicolaus this should be fine, as long as the server that you load the config.json file from supports CORS properly. I believe "github pages" sites allow CORS.

so from jbrowse2's perspective, using something like

http://yoursite.com/jbrowse2/?config=http://someothersite/config.json (where someother site would probably be a yourusername.github.io/ in the github pages case)

that is fine

you could also deploy jbrowse entirely on e.g. github pages. this is also allowed, as jbrowse2 does not technically need a server side, but it just depends on how you set up things

**Reply by johannesnicolaus** [reply]
Hi thanks so much for the reply. I was thinking that maybe if I could have the config set by default to take URL so that i could just have something like `http://mysite.com` without the `/?config` part. If that's not possible then maybe i can just redirect by default. Sorry if my understanding is limited, quite new to this...

**Reply by cmdcolin** [reply]
gotcha, there actually is not a way to avoid using the ?config currently afaik with the stock jbrowse-web app. 

we could probably make it so that you could configure the config URL that is fetched by default though

alternatively: this is a step up in complexity but you can use @jbrowse/react-app which actually makes you fetch your own config.json type data. it lacks a couple features from jbrowse-web has like session sharing button but is otherwise fairly similar to jbrowse-web https://jbrowse.org/storybook/app/main/?path=/docs/getting-started--docs

**Reply by cmdcolin** [reply]
the @jbrowse/react-app is generally geared for users that want maximum control over programmatically controlling things and/or npm-installability. if you don't need that, it may not be necessary to go down that road, and we could likely find a way to make jbrowse-web accept some sort of global variable for the config URL

**Reply by johannesnicolaus** [reply]
I see, that's understandable. maybe this could be implemented in the future?

**Reply by cmdcolin** [reply]
I think so, I made a feature request here https://github.com/GMOD/jbrowse-components/issues/4012

**Comment by cmdcolin** [comment, marked as answer]
this was now added in v2.8.0, it is a little cumbersome but if you manually edit the index.html of the jbrowse-web instance to have something like this

```html
<script>
window.__jbrowseConfigPath="...your config.json path here"
</script>
```

then it will be used

**Reply by johannesnicolaus** [reply]
This is great, thanks so much!
